rust/pact_matching/src/models/mod.rs

```diff
@@ -1784,46 +1784,41 @@ pub fn message_interaction_from_json(source: &str, json: &Value, spec: &PactSpec
 }
 
 /// Reads the pact file and parses the resulting JSON into a `Pact` struct
-pub fn read_pact(file: &Path) -> io::Result<Box<dyn Pact>> {
+pub fn read_pact(file: &Path) -> anyhow::Result<Box<dyn Pact>> {
   let mut f = File::open(file)?;
   read_pact_from_file(&mut f, file)
 }
 
 /// Reads the pact from the file and parses the resulting JSON into a `Pact` struct
-pub fn read_pact_from_file(file: &mut File, path: &Path) -> io::Result<Box<dyn Pact>> {
+pub fn read_pact_from_file(file: &mut File, path: &Path) -> anyhow::Result<Box<dyn Pact>> {
   let buf = with_read_lock_for_open_file(path, file, 3, &mut |f| {
     let mut buf = String::new();
     f.read_to_string(&mut buf)?;
     Ok(buf)
   })?;
-  let pact_json = serde_json::from_str(&buf);
-  match pact_json {
-    Ok(ref json) => load_pact_from_json(&*path.to_string_lossy(), json)
-      .map_err(|err| Error::new(ErrorKind::Other, err)),
-    Err(err) => {
+  let pact_json = serde_json::from_str(&buf)
+    .with_context(|| "Failed to parse Pact JSON")
```

Comment:
Since this is just a string literal, probably replace with `.context("Failed to parse Pact JSON")`

Reply:
I had thought about this; the idea I had was that it *might* be a dynamically formatted string in the future to add more context. But I suppose it's easy enough to change to `with_context` if someone adds that :)